app/lib/primer/test_selector_helper.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Primer
+  # Module to allow components to deal with the `test_selector` argument.
+  # It will only add the selector if env is not Production.
+  #
+  # test_selecotr: "foo" => data-test-selector="foo"
+  module TestSelectorHelper
+    TEST_SELECTOR_TAG = :test_selector
+
+    def add_test_selector(args)
+      if args.key?(TEST_SELECTOR_TAG) && !Rails.env.production?
+        args[:data] ||= {}
+        args[:data][TEST_SELECTOR_TAG] = args[TEST_SELECTOR_TAG]
+      end
+
+      args.except(TEST_SELECTOR_TAG)
```

Comment:
I've found this method confusing in that it's called `add_test_selector` but it actually removes it in all envs. I had started poking at this in https://github.com/primer/view_components/pull/275. I don't have a particular recommendation here but curious if any of these sparks thoughts for you.